version: "3"

services:
  pihole:
    container_name: pihole
    image: pihole/pihole:latest
    ports:
      # https://forums.docker.com/t/dns-issues-with-local-resolver-and-containers-on-the-same-host/102319/4
      # not 100% sure why binding the host's IP address is required here
      - "${IP4:?err}:53:53/tcp"
      - "${IP4:?err}:53:53/udp"
      - "${IP4:?err}:67:67/udp"
    environment:
      - 'TZ=America/New_York'
    labels:
      - "traefik.enable=true"
      - "traefik.http.routers.pihole.rule=Host(`pihole.${PIHOME_HOSTNAME:?err}.${PIHOME_TLD:?err}`)"
      - "traefik.http.routers.pihole.entrypoints=websecure"
      - "traefik.http.routers.pihole.tls=true"
      - "traefik.http.services.pihole.loadbalancer.server.port=80"
      - "traefik.docker.network=traefik-net"
      - "traefik.http.middlewares.pihole-add-admin-path.redirectregex.regex=^https://pihole.${PIHOME_HOSTNAME:?err}.${PIHOME_TLD:?err}/?$"
      - "traefik.http.middlewares.pihole-add-admin-path.redirectregex.replacement=https://pihole.${PIHOME_HOSTNAME:?err}.${PIHOME_TLD:?err}/admin"
      - "traefik.http.middlewares.pihole-add-admin-path.redirectregex.permanent=true"
      - "traefik.http.routers.pihole.middlewares=pihole-add-admin-path"
    volumes:
       - './etc-pihole/:/etc/pihole'
       - './etc-dnsmasq.d/:/etc/dnsmasq.d'
    networks:
      traefik-net:
        ipv4_address: 172.22.0.11
    restart: unless-stopped
networks:
  traefik-net:
    external: true
